src/coreclr/jit/importercalls.cpp

```diff
@@ -3176,6 +3179,20 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                             retNode = gtNewIconNode(
                                 (info.compCompHnd->getClassAttribs(hClass) & CORINFO_FLG_BYREF_LIKE) ? 1 : 0);
                             break;
+                        case NI_System_Type_get_IsPrimitive:
+                            // getTypeForPrimitiveValueClass returns underlying type for enums, so we check it first
+                            // because enums are not primitive types.
+                            if ((info.compCompHnd->isEnum(hClass, nullptr) == TypeCompareState::MustNot) &&
```

Comment:
Should the intrinsic avoid expanding for `TypeCompareState::May`?

Reply:
> Should the intrinsic avoid expanding for `TypeCompareState::May`?

if it returns May then it's definitely not a primitive